internal/commands/user/shared/flags.go

```diff
@@ -53,3 +66,51 @@ func IsValidProviderType(pt string) bool {
 	}
 	return false
 }
+
+// StatusType enumset
+type StatusType string
+
+// String returns the status type display
+func (st StatusType) String() string { return string(st) }
+
+// Type returns the StatusType type
+func (st StatusType) Type() string { return "string" }
+
+// Set validates and sets the status type value
+func (st *StatusType) Set(val string) error {
+	newStatusType := StatusType(val)
+
+	if !IsValidStatusType(newStatusType) {
+		return ErrInvalidStatusType
+	}
+
+	*st = newStatusType
+	return nil
+}
+
+// set of StatusType(s)
+const (
+	StatusTypeNil       StatusType = ""
```

Comment:
[nit] can we make this StatusTypeEmpty? Empty is more idiomatic go since nil denotes that it is truly a nil value, not an empty one. 

I know you moved this code but first time I'm seeing it. @makesitgo is this even necessary? I'm leaning towards a `status == ""` is a lot cleaner and more straight forward than explicitly having this empty status type.

Reply:
Totally fine with using the empty terminology here. @makesitgo